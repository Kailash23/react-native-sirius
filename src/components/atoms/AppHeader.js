import React from 'react';
import { View, Image, StatusBar, Platform } from 'react-native';
import { withTheme } from 'react-native-paper';
import { SIZES } from '@styles/sizes';
import { withNavigation } from '@react-navigation/compat';
import { useMappedState } from 'redux-react-hook';

const mappedState = state => ({
  selectedTheme: state.sessionReducer.selectedTheme
});

function AppHeader(props) {
  const { theme } = props;
  const { selectedTheme } = useMappedState(mappedState);
  let logo =
    selectedTheme === 'light'
      ? require('../../../react/images/logo.png')
      : require('../../../react/images/logo_dark_mode_.png');

  return (
    <View
      style={[
        styles.container,
        {
          backgroundColor: theme.colors.headerBackground
        },
        Platform.OS === 'ios' && styles.iosStyle,
        Platform.OS === 'android' && styles.androidStyle
      ]}>
      <StatusBar
        barStyle="dark-content"
        backgroundColor={theme.colors.headerBackground}
      />
      <Image
        resizeMode={Platform.OS === 'ios' ? 'contain' : 'center'}
        source={logo}
        style={styles.imageStyle}
      />
    </View>
  );
}

const styles = {
  iosStyle: {
    flex: 1
  },
  androidStyle: {
    height: SIZES.LOGO_HEIGHT,
    width: '100%'
  },
  container: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignSelf: 'center'
  },
  imageStyle: {
    width: SIZES.LOGO_WIDTH,
    alignContent: 'center',
    height: SIZES.LOGO_HEIGHT,
    resizeMode: 'contain',
    position: 'absolute',
    top: -1,
    backgroundColor: 'transparent'
  }
};

export default withTheme(withNavigation(AppHeader));
