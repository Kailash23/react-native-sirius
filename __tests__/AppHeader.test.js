import AppHeader from '@atoms/AppHeader';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';

configure({ adapter: new Adapter() });

describe('Testing <AppHeader/>', () => {
  test('should render a AppHeader', () => {
    const wrapper = shallow(<AppHeader />);
    expect(wrapper).toMatchSnapshot();
  });
});
