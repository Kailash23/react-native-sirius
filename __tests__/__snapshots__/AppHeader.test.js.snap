// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <AppHeader/> should render a AppHeader 1`] = `
<ThemedComponent
  _reactThemeProviderForwardedRef={null}
/>
`;
